gramineproject/gramine
# [Pal/Linux-SGX] Add MAA to RA-TLS and SecretProv libs

## Description of the changes 

Microsoft Azure Attestation (MAA) is a new attestation scheme available in the MS Azure public cloud. It uses the classic DCAP SGX quotes but replaces the DCAP/EPID quote-verification software infrastructure with the new MAA attestation provider infrastructure. In particular, the remote user is supposed to send the SGX quote in the HTTPS REST attestation request to the MAA attestation provider and receive back the JSON Web Token (JWT) that contains claims about the attesting SGX enclave.

This commit introduces new `_maa` versions of the RA-TLS and Secret Provisioning verification libraries. The documentation and the `ra-tls-mbedtls` and `ra-tls-secret-prov` examples are updated.

For design, see https://github.com/gramineproject/gramine/issues/626.

This PR does *not* depend on any other PRs, but as part of this work I detected and fixed some attestation-related issues:
- https://github.com/gramineproject/gramine/pull/638
- https://github.com/gramineproject/gramine/pull/643
- https://github.com/gramineproject/gramine/pull/645
- https://github.com/gramineproject/gramine/pull/647

## How to test this PR? 

Manually verified on a MS Azure CC VM, with the `https://sharedcus.cus.attest.azure.net` default attestation provider. See the READMEs of `ra-tls-mbedtls` and `ra-tls-secret-prov` examples.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/652)
